Fix related products E2Es for WC9.5

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

This PR adjusts E2E tests for WC 9.5 changes.

In WC 9.5, by default, there is no related products section on a single product page - I made the test skip in such a case.
Also, I added a "related" tag to all simple products to make sure simple products will appear in the related products section instead of "Cat teasers" - without the "Add to cart" button.



### Screenshots:

![image](https://github.com/user-attachments/assets/7c9a774c-2b82-4421-8184-0c014d0434d2)


<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Run E2E tests locally
2. Run [E2E workflow](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/workflows/e2e-tests.yml) with `9.5.0-rc.1` - https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12362867871
3. Run [E2E workflow](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/workflows/e2e-tests.yml) with `9.4.0` - https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12383008805
4. Review the code to check the tests are testing what they should


### Additional details:

- I'll make a separate PR with a proposal to simplify and DRY a bit our `create*Page`s

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Dev - Improve E2E tests for related products.

tomalec committed:
Skip when there is no related products section

tomalec committed:
Add dedicated test for related products block.

eason9487 commented:
> 3\. Run [E2E workflow](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/workflows/e2e-tests.yml) with `9.4.0` - https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12363338232

This run didn't pass. I got the same failure in a local run.

tomalec committed:
Update add to cart locator to match product collection block

tomalec committed:
Fix even data retrieval for related products test

tomalec commented:
Thanks. I forgot to migrate the adjusted selector. Now, the run is passing https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12383008805

eason9487 commented on tests/e2e/utils/create-page.js:
```diff
@@ -135,3 +136,18 @@ export async function createAllProductsBlockShopPage() {
+ * Creates a page with related products block.
+ *
+ * @param {number} productId Product ID.
+ * @return {string} Slug of the created page.
```
```suggestion
 * @return {Promise<string>} Slug of the created page.
```

eason9487 reviewed (approved):
Thanks for the compatibility fix! LGTM.

eason9487 commented:
Hi @tomalec, I will address https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/481#discussion_r1889560287 and merge this PR to make the later release process a bit smoother.

eason9487 committed:
Fix type declaration in JSDoc.

Address https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/481#discussion_r1889560287
